src/hotspot/share/services/virtualMemoryTracker.cpp

```diff
@@ -107,6 +108,25 @@ static bool try_merge_with(LinkedListNode<CommittedMemoryRegion>* node, LinkedLi
   return try_merge_with(node, rgn->base(), rgn->size(), *rgn->call_stack());
 }
 
+ReservedMemoryRegion::ReservedMemoryRegion(address base, size_t size, const NativeCallStack& stack, MEMFLAGS flag) :
+  VirtualMemoryRegion(base, size),
+  _committed_regions(VirtualMemoryTracker::_backing_arena),
+  _stack(stack), _flag(flag) { }
+
+
+ReservedMemoryRegion::ReservedMemoryRegion(address base, size_t size) :
+  VirtualMemoryRegion(base, size),
+  _committed_regions(VirtualMemoryTracker::_backing_arena),
+  _stack(NativeCallStack::empty_stack()), _flag(mtNone) { }
+
+// Copy constructor
+ReservedMemoryRegion::ReservedMemoryRegion(const ReservedMemoryRegion& rr) :
+    VirtualMemoryRegion(rr.base(), rr.size()),
+    _committed_regions(VirtualMemoryTracker::_backing_arena) {
+    *this = rr;
+}
```

Comment:
(preexisting) since you have a copy constructor, should you have a deleted operator=?

Reply:
Yeah, I also thought it was weird that Rule-of-3 isn't obeyed here. Just at a glance, seems like the default destructor does the right thing (calling destructor for each member).